src/vm/script.rs

```diff
@@ -321,6 +321,344 @@ impl Script {
                             }
                         }
 
+                        ScriptExecutorState::ExpectingBytesOrCachedTerm(OP::Unsigned16Var) => {
+                            let mut failed = false;
+                            let mut arr: [u8; 2] = [0; 2];
+
+                            for j in 0..2 {
+                                frame.i_ptr += 1;
+                                let i = &f.script[frame.i_ptr];
+
+                                match i {
+                                    ScriptEntry::Byte(byte) => {
+                                        arr[j] = *byte;
```

Comment:
Since this is already validated by the parser, it is faster to do this (in case of u16):
```rust
let b = byte as u16;
if *j == 0 {
    *sum += b;
} else {
    *sum += b << 8;
}
```

This would be in the case of the others:
```rust
let b = byte as u32;
*sum += b << 8 * j;
```

In case of signed integers, we decode them the same as unisgned integers and then we can cast the binary representation:
```rust
let sum = unsafe {
    mem::transmute::<u32, i32>(*sum)
};
```

I would still benchmark to confirm

Reply:
Done :)